apps/queries/api.py

```diff
@@ -203,83 +203,150 @@ def get(self, request, **kwargs):
             return Response(dict_fetch_all(cursor), status=status.HTTP_200_OK)
 
 
+user_categories = [
+    {"prefix": "ins", "permission": "institution", "weight": 1, "identifier": "institution_id"},
+    {"prefix": "f", "permission": "faculty", "weight": 2, "identifier": "faculty_id"},
+    {"prefix": "i", "permission": "issuer", "weight": 3, "identifier": "issuer_id"},
+    {"prefix": "bc", "permission": "badge_class", "weight": 4, "identifier": "badgeclass_id"}
+]
+
+
 class Users(APIView):
     permission_classes = (TeachPermission,)
 
-    permission_badge_class = "badge_class"
-    permission_issuer = "issuer"
-    permission_faculty = "faculty"
-    permission_institution = "institution"
-
-    admin = "admin"
-    editor = "editor"
-    awarder = "awarder"
+    @staticmethod
+    def _remove_duplicate_permissions(user):
```

Comment:
I do not understand this change completely. You check whether the use has duplicate permissions (what are duplicate permissions?) and if so, make them unique again? So this "clean up" happens every time a GET user is done?

Reply:
Because of the left joins in the query, some permissions are duplicated. Those ones are removed before send back to the client. Uniqueness is based on the identifier (e.g. institution_id, ) and permission level. Note that the user parameter is just a row from the database, not an actual user object. I'll change the name of the parameter to avoid confusing.

Reply:
https://github.com/edubadges/edubadges-server/pull/172/commits/f19705d88607a64fe7dd279f54818814573e8d1a